src/modules/core/utils/time.utils.ts

```diff
@@ -0,0 +1,105 @@
+type TimeUnit = `${number}${number}`;
+type TimeValue = `${TimeUnit}:${TimeUnit}`;
+type TimeObject = { hours: number; minutes: number };
+
+export class Time {
+  private value: TimeValue;
+  constructor(value?: Time | TimeValue | TimeObject) {
+    if (!value) {
+      this.value = '00:00';
+
+      return;
+    }
+
+    if (value instanceof Time) {
+      this.value = value.getString();
+
+      return;
+    }
+
+    if (typeof value === 'object') {
+      this.value = new Time().set(value).getString();
+
+      return;
+    }
+
+    this.value = value;
+  }
+
+  public set(value: Partial<TimeObject> | TimeValue): this {
+    if (typeof value === 'object') {
+      const time = {
+        ...this.getTimeObject(),
+        ...value,
+      };
+
+      const hours = Time.timeUnitToString(time.hours);
+      const minutes = Time.timeUnitToString(time.minutes);
+
+      this.value = `${hours}:${minutes}`;
+
+      return this;
+    }
+
+    this.value = value;
+
+    return this;
+  }
+
+  public getHours(): number {
+    const [h] = this.value.split(':');
+
+    return +h;
+  }
+
+  public getMinutes(): number {
+    const [, m] = this.value.split(':');
+
+    return +m;
+  }
+
+  public getTimeObject(): TimeObject {
+    const [h, m] = this.value.split(':');
+
+    return {
+      hours: +h,
+      minutes: +m,
+    };
+  }
+
+  public getString(): TimeValue {
+    return this.value;
+  }
+
+  public static timeUnitToString(unit: number): `${number}${number}` {
+    const MAX_NUMERIC = 9;
+
+    return unit <= MAX_NUMERIC ? `0${unit}` : (unit.toString() as TimeUnit);
+  }
+
+  public static guard(value: string): value is TimeValue {
+    const timeRegex = /^(?:[01]\d|2[0-3]):[0-5]\d$/;
+
+    return timeRegex.test(value);
+  }
+}
+
+export const formatTimeRange = (startTime: Time, endTime: Time): string => {
```

Comment:
Would be nice if you add JSDoc for all utils, so we have nice documentation in our IDE.